lib/services/platform-service.ts

```diff
@@ -368,15 +368,56 @@ export class PlatformService implements IPlatformService {
 				this.$errors.failWithoutHelp("Unable to find built application. Try 'tns build %s'.", platform);
 			}
 
-			this.$fs.ensureDirectoryExists(path.dirname(targetPath)).wait();
+			function ext(file: string): string {
+				return path.extname(file).toLowerCase();
+			}
+			function targetExt(file: string): string {
+				let e = ext(file);
+				return (e === ".ipa" || e === ".app") ? e : ".";
```

Comment:
If it is not ".ipa" or ".app" it will be treated as directory path. I do not like the `.` either but couldn't figure out better while I was writing it. Other options are `*`, `?`, `<dir>`, `(dir)` etc. Would you prefer one of these?